### PR TITLE
Add Bicep Deployment 

### PR DESCRIPTION
**Pull request by gjoshevski**

### CODE REVIEW

**Comment by chgeuer** [comment]
1. I like the idea of having multiple samples on how to send usage messages into the system (into EventHub), but I certainly would not put a Python demo into the `deploy/README.md` file (as IMO it does not really belong to deploying a production environment). 
2. In the `deploy/README.md`, the sample shell commands calling `az` refer to `CHANGE-WITH-CLIENT_ID` variables, whereas it isn't obvious whether this refers to infrastructure credentials (to talk to EventHub or storage), or whether this is the marketplace credential. I guess it's the latter. In the current main [Readme](https://github.com/microsoft/metered-billing-accelerator/blob/main/README.md#azure-marketplace-api-credential-to-submit-metering-values-to-azure), there are already environment vars (called `AZURE_METERING_MARKETPLACE_CLIENT_ID`, `AZURE_METERING_MARKETPLACE_CLIENT_SECRET` and `AZURE_METERING_MARKETPLACE_TENANT_ID`, so I'd certainly use these for consistency

File: deploy/main-existing-rg.bicep [hunk 1]
```diff
@@ -0,0 +1,165 @@
+// name should not contain underscores
+
+@description('Prefix for created resources.')
+param appNamePrefix string
+
+@allowed([
+  'northcentralusstage'
+  'westcentralus,eastus'
```
Comment (chgeuer) [inline deploy/main-existing-rg.bicep:8]: This doesn't look right, should that be 2 lines?

File: deploy/main-existing-rg.bicep [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+])
+param skuCapacity int = 1
+param consumerGroupName string = 'consumerGroupName'
+
+
+
+//Container App params 
+param minReplicas int = 1
+
+param containerImage string = 'ghcr.io/gjoshevski/metered-billing-accelerator:main'
```
Comment (chgeuer) [inline deploy/main-existing-rg.bicep:41]: Is that supposed to be `'ghcr.io/gjoshevski`? Can we use microsoft here?

File: deploy/main-existing-rg.bicep [hunk 3]
```diff
@@ -0,0 +1,165 @@
+// name should not contain underscores
+
+@description('Prefix for created resources.')
+param appNamePrefix string
+
+@allowed([
```
Comment (chgeuer) [inline deploy/main-existing-rg.bicep:6]: Generally speaking, which resource puts the location constraint in? What is only avail in these regions? Maybe put in a comment where that comes from

File: deploy/main-new-rg.bicep [hunk 4]
```diff
@@ -0,0 +1,37 @@
+targetScope = 'subscription'
+param appNamePrefix string
+@allowed([
```
Comment (chgeuer) [inline deploy/main-new-rg.bicep:3]: Same here. What's only avail in there regions? Container Apps?

File: deploy/modules/role-assignments.bicep [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+resource eventHub 'Microsoft.EventHub/namespaces/eventhubs@2021-11-01' existing = {
+  name: eventHubName
+}
+
+resource eventhubRoleAssignment 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
+  name: guid(appNamePrefix, eventHubName)
+
+  scope: eventHub
+  properties: {
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'f526a384-b230-433a-b45c-95f59c4a2dec')
```
Comment (chgeuer) [inline deploy/modules/role-assignments.bicep:18]: I would move `'f526a384-b230-433a-b45c-95f59c4a2dec'` into a named variable 

```bicep
var azureEventHubsDataOwnerRole = 'f526a384-b230-433a-b45c-95f59c4a2dec'
```

**Review by chgeuer — APPROVED** [review]

**Comment by gjoshevski** [comment]
@chgeuer Thanks for your review: 

I have accepted all your suggestions the only remaining things are:

1.  The var names used in the deployment will remain the same, since the current params (tenantID, ADApplicationID, ADApplicationSecret) are used in [Marketplace-SaaS-Accelerator](the https://github.com/Azure/Commercial-Marketplace-SaaS-Accelerator/blob/main/docs/Installation-Instructions.md) and I believe this will provide more unified UX for customers that use both of the solution together. 

2.  The image repository will remain as is, until we make the one under microsoft organization public.

**Pull request merged**